## Q1. What are the three measures of central tendency?

Ans:
The three measures of central tendency are:

1. Mean: This is the sum of all the values in a dataset divided by the total number of values. It is also called the arithmetic mean.

2. Median: This is the middle value in a dataset when the values are arranged in order. If there is an even number of values, then the median is the average of the two middle values.

3. Mode: This is the value that appears most frequently in a dataset. A dataset can have one mode (unimodal), two modes (bimodal), or more than two modes (multimodal)

## Q2. What is the difference between the mean, median, and mode? How are they used to measure the central tendency of a dataset?

Ans:

The mean, median, and mode are all measures of central tendency, but they differ in how they are calculated and what they represent.


1. The mean is calculated by adding up all the values in a dataset and dividing by the total number of values.The median is the middle value in a dataset when the values are arranged in order. It represents the value that splits the dataset into two equal halves. The mode is the value that appears most frequently in a dataset.

2. Mean is sensitive to outliers. Median is less sensitive to outliers. Mode is insensitive to outliers.

To measure central tendency of a dataset: 'Mean' is used when data is symmetrical and has no outliers.If the dataset is skewed or has outliers, the median may be a better measure of central tendency. If the dataset has several distinct peaks, the mode may be a good measure of central tendency

## Q3. Measure the three measures of central tendency for the given height data:

 ## [178,177,176,177,178.2,178,175,179,180,175,178.9,176.2,177,172.5,178,176.5]

In [6]:
import numpy as np      
from statistics import mode
data = [178, 177, 176, 177, 178.2, 178, 175, 179, 180, 175, 178.9, 176.2, 177, 172.5, 178, 176.5]
mean = np.mean(data)
median = np.median(data)
mode= mode(data)
print("Mean height:", mean)
print("Median height:", median)
print("Mode height:", mode)

Mean height: 177.01875
Median height: 177.0
Mode height: 178


## Q4. Find the standard deviation for the given data:

## [178,177,176,177,178.2,178,175,179,180,175,178.9,176.2,177,172.5,178,176.5]

In [9]:
import numpy as np
data = [178, 177, 176, 177, 178.2, 178, 175, 179, 180, 175, 178.9, 176.2, 177, 172.5, 178, 176.5]
std_height = np.std(data)
print("Standard deviation:", std_height)


Standard deviation: 1.7885814036548633


## Q5. How are measures of dispersion such as range, variance, and standard deviation used to describe the spread of a dataset? Provide an example.

Ans:

Measures of dispersion, such as range, variance, and standard deviation, are used to describe how spread out the values in a dataset are. These measures provide information about the variability or diversity of the data points in a dataset.

For example, let's say we have two datasets:
Dataset 1: [2, 2, 3, 4, 5]
Dataset 2: [3, 1, 4, 4, 4]

In [14]:
import numpy as np
data1 = [1, 2, 3, 4, 5]
data2 = [2, 1, 4, 4, 4]
#mean
mean1=np.mean(data1)
mean2=np.mean(data2)
# Range
range1 = np.max(data1) - np.min(data1)
range2 = np.max(data2) - np.min(data2)
# Variance
var1 = np.var(data1)
var2 = np.var(data2)
# Standard deviation
std1 = np.std(data1)
std2 = np.std(data2)
# Print results
print("Dataset 1")
print("Mean1:", mean1)
print("Range:", range1)
print("Variance:", var1)
print("Standard deviation:", std1)
print()
print("Dataset 2")
print("mean2:", mean2)
print("Range:", range2)
print("Variance:", var2)
print("Standard deviation:", std2)


Dataset 1
Mean1: 3.0
Range: 4
Variance: 2.0
Standard deviation: 1.4142135623730951

Dataset 2
mean2: 3.0
Range: 3
Variance: 1.6
Standard deviation: 1.2649110640673518


Both datasets have the same mean of 3, but the values in the two datasets are distributed differently. We can use measures of dispersion to describe the spread of the data in each dataset.


The range of a dataset is the difference between the largest and smallest values. For Dataset 1, the range is 5 - 1 = 4, while for Dataset 2, the range is 4 - 2 = 2.

The range gives us a quick and simple way to compare the spread of the two datasets.


The variance and standard deviation measure the average distance of each data point from the mean.

For Dataset 1, the variance is 2 and the standard deviation is approximately 1.41. 
For Dataset 2, the variance is 1.6 and the standard deviation is approximately 1.26. 
The smaller the variance or standard deviation, the less spread out the data is around the mean.

In this example, we can see that while both datasets have the same mean, they have different ranges, variances, and standard deviations, indicating different levels of spread or variability. The measures of dispersion allow us to quantify and compare the spread of the data in different datasets, helping us to better understand and analyze the data

## Q6. What is a Venn diagram?

Ans:
    
A Venn diagram is a graphical representation of the relationships between sets. It is composed of a series of overlapping circles or ellipses, with each circle representing a set and the overlap representing the intersection between those sets.    

## Q7. For the two given sets A = (2,3,4,5,6,7) & B = (0,2,6,8,10). Find:

## (i) 	A ⋂ B
## (ii)	A ⋃ B

In [16]:
A = {2, 3, 4, 5, 6, 7}
B = {0, 2, 6, 8, 10}
intersection = A.intersection(B)
print("A ⋂ B =", intersection)
union = A.union(B)
print("A ⋃ B =", union)


A ⋂ B = {2, 6}
A ⋃ B = {0, 2, 3, 4, 5, 6, 7, 8, 10}


## Q8. What do you understand about skewness in data?

Ans:

Skewness is a statistical measure that describes the degree of asymmetry in a probability distribution. A distribution is said to be skewed if it is not symmetric, i.e., if one tail of the distribution has more mass than the other tail.

There are three types of skewness:
1. Positive skewness: In a positively skewed distribution, the tail on the right-hand side of the distribution is longer or has more mass than the tail on the left-hand side.

2. Negative skewness: In a negatively skewed distribution, the tail on the left-hand side of the distribution is longer or has more mass than the tail on the right-hand side.

3. Zero skewness: In a symmetric distribution, the two tails are of equal length or have equal mass.


Skewness is important in data analysis because it can affect the interpretation of statistics and the validity of certain statistical tests. For example, if the data is skewed, the mean may not be an accurate representation of the central tendency of the data, and other measures of central tendency, such as the median or mode, may be more appropriate. Similarly, if the data is skewed, it may violate the assumptions of certain statistical tests, such as the t-test, and alternative tests may need to be used.

## Q9. If a data is right skewed then what will be the position of median with respect to mean?

Ans:
    
If a dataset is right-skewed, the median will be less than the mean. This is because in a right-skewed distribution, the tail on the right-hand side is longer or has more mass than the tail on the left-hand side

## Q10. Explain the difference between covariance and correlation. How are these measures used in statistical analysis?

Ans:
    
Covariance and correlation are two measures used to describe the relationship between two variables in a dataset.

1. Covariance measures how much two variables change together. It is a measure of the joint variability of two random variables, i.e., how much they move together in a linear fashion. It can take on both positive and negative values.A positive covariance indicates that the two variables tend to increase or decrease together, while a negative covariance indicates that one variable tends to increase while the other decreases. However, covariance does not provide information about the strength and direction of the relationship between two variables.

2. Correlation, on the other hand, is a standardized measure that describes the strength and direction of the linear relationship between two variables. It measures how closely two variables move together and ranges from -1 to +1. A correlation coefficient of +1 indicates a perfect positive correlation, meaning that the two variables move in the same direction in a linear fashion, while a correlation coefficient of -1 indicates a perfect negative correlation, meaning that the two variables move in opposite directions in a linear fashion. A correlation coefficient of 0 indicates no linear relationship between the two variables.


Correlation is preferred over covariance when comparing the relationship between variables because it is normalized, which means it is not affected by the scales of the two variables. For example, if one variable is measured in dollars and the other in euros, covariance may not provide an accurate measure of the relationship between the two variables. Correlation, on the other hand, is scale-invariant and provides a more accurate measure of the relationship between the two variables.

## Q11. What is the formula for calculating the sample mean? Provide an example calculation for a dataset.

Ans:
    
The formula for calculating the sample mean is:
   
sample mean = (sum of all data values)/(number of data values)

In [17]:
#python example
data = [12, 15, 18, 21, 24]
n = len(data)
mean = sum(data) / n
print("Sample mean:", mean)


Sample mean: 18.0


## Q12. For a normal distribution data what is the relationship between its measure of central tendency?

Ans:
    
For a normal distribution data, the mean, median, and mode are all equal. This is because the normal distribution is symmetric around its mean, which means that the middle value (median) is the same as the peak value (mode) and also the same as the arithmetic average (mean).

## Q13. How is covariance different from correlation?

Ans:
    
1. Covariance indicates the direction and strength of the linear relationship between two variables, while correlation measures the strength of the linear relationship on a standardized scale from -1 to 1. 

2. Covariance is difficult to interpret because it depends on the units of the variables. On the other hand,correlation is unitless and therefore easier to interpret and compare across different datasets.

3. A positive covariance indicates that when one variable increases, the other variable tends to increase as well, and when one variable decreases, the other variable tends to decrease as well. A negative covariance indicates that when one variable increases, the other variable tends to decrease and vice versa.
On the other hand,a correlation of +1 indicates a perfect positive relationship, a correlation of -1 indicates a perfect negative relationship

## Q14. How do outliers affect measures of central tendency and dispersion? Provide an example.

Ans:
    
Outliers have a significant effect on measures of central tendency and dispersion.

1. Affect on central tendency:
If a dataset has outliers, the mean will be pulled towards the direction of the outliers, making it an unreliable measure of central tendency. The median is less affected by outliers, but can still be skewed if there are multiple outliers. The mode may not be affected by outliers at all.

2. Affect on dispersion: The range will increased if there are outliers on either end of the dataset. The variance and standard deviation will also be increased, as they are based on the squared deviations from the mean, which will be larger if there are outliers in the dataset.

In [19]:
#Example
import numpy as np
# dataset with an outlier
data = np.array([40, 45, 50, 55, 60, 65, 70, 75, 80, 85, 90, 95, 100, 1000000])
# calculate the mean, median, and standard deviation
mean = np.mean(data)
median = np.median(data)
std_dev = np.std(data)
print("Mean: ", mean)
print("Median: ", median)
print("Standard Deviation: ", std_dev)


Mean:  71493.57142857143
Median:  72.5
Standard Deviation:  257521.34969349467


As we can see, the mean salary is pulled towards the outlier value of 1,000,000, resulting in a mean salary of 76970.5 which is not a representative measure of the central tendency. On the other hand, the median is not affected by the outlier, resulting in a median salary of 75. The standard deviation is also significantly larger due to the presence of the outlier.